DependencyInjection/MonologExtension.php

```diff
@@ -668,6 +668,51 @@ private function buildHandler(ContainerBuilder $container, $name, array $handler
             ));
             break;
 
+        case 'sentry':
+            if (null !== $handler['client_id']) {
+                $clientId = $handler['client_id'];
+            } else {
+                $options = new Definition(
+                    'Sentry\\Options',
+                    array(array('dsn' => $handler['dsn']))
+                );
+
+                if (!empty($handler['environment'])) {
+                    $options->addMethodCall('setEnvironment', [$handler['environment']]);
+                }
+
+                if (!empty($handler['release'])) {
+                    $options->addMethodCall('setRelease', [$handler['release']]);
+                }
+
+                $builder = new Definition('Sentry\\ClientBuilder', array($options));
+
+                $client = new Definition('Sentry\\Client');
+                $client->setFactory(array($builder, 'getClient'));
+
+                $clientId = 'monolog.sentry.client.'.sha1($handler['dsn']);
+                $container->setDefinition($clientId, $client);
+
+                if (!$container->hasAlias('Sentry\\ClientInterface')) {
+                    $container->setAlias('Sentry\\ClientInterface', $clientId);
+                }
+            }
+
+            $hub = new Definition(
+                'Sentry\\State\\Hub',
+                array(new Reference($clientId))
+            );
+
+            // can't set the hub to the current hub, getting into a recursion otherwise...
```

Comment:
It's just a singleton. You could use a factory and use the `getCurrent` method to obtain it, but I'm not sure if it would collide with the Sentry Bundle; order of setup of bundles would make a mess here.

Reply:
As far as I can tell, the singleton part is for the helper functions. I don't think we would want to use these if we're opting for monolog.

With what I made, I think that you can use both (it's just that the hub will be instantiated two times under differents ids, and have the singleton instantiated in the sentry bundle... especially as having both is kinda missing the point IMO).